packages/react-components/react-dialog/e2e/Dialog.e2e.tsx

```diff
@@ -135,14 +140,55 @@ describe('Dialog', () => {
     cy.get(dialogTriggerCloseSelector).click();
     cy.get(dialogTriggerOpenSelector).should('be.focused');
   });
+  it('should allow change of focus on open', () => {
+    const CustomFocusedElementOnOpen = () => {
+      const buttonRef = React.useRef<HTMLButtonElement>(null);
+      const [open, setOpen] = React.useState(false);
+      React.useEffect(() => {
+        if (open && buttonRef.current) {
+          buttonRef.current.focus();
+        }
+      }, [open]);
+      return (
+        <Dialog open={open} onOpenChange={(event, data) => setOpen(data.open)}>
+          <DialogTrigger>
+            <Button>Open dialog</Button>
+          </DialogTrigger>
+          <DialogSurface>
+            <DialogTitle>Dialog title</DialogTitle>
+            <DialogBody>This dialog focus on the second button instead of the first</DialogBody>
+            <DialogActions position="start">
+              <Button appearance="outline">Third Action</Button>
+            </DialogActions>
+            <DialogActions position="end">
+              <DialogTrigger>
+                <Button id={dialogTriggerCloseId} ref={buttonRef} appearance="secondary">
+                  Close
+                </Button>
+              </DialogTrigger>
+              <Button appearance="primary">Do Something</Button>
+            </DialogActions>
+          </DialogSurface>
+        </Dialog>
+      );
+    };
+    mount(<CustomFocusedElementOnOpen />);
+    cy.get(dialogTriggerOpenSelector).click();
+    cy.get(dialogTriggerCloseSelector).should('be.focused');
+  });
   describe('modalType = modal', () => {
     it('should close with escape keydown', () => {
       mount(
         <Dialog modalType="modal">
           <DialogTrigger>
             <Button>Open dialog</Button>
           </DialogTrigger>
-          <DialogSurface>
+          {/*
+              Cypress doesn't actually simulate a keyboard press.
+              Instead it fires a bunch of events manually to simulate keypress,
+              Which will not fire `cancel` event on `<dialog>`
+          */}
```

Comment:
you can use `realPress` to use cdp to trigger real events

Reply:
This will actually only be a problem on native `dialog`, so for now I'll just remove this comment, as it's working fine with a `div`